Remotion/Validation/UnitTests/Attributes/Validation/NotEmptyValidationAttributeTest.cs

```diff
@@ -40,35 +40,99 @@ public void SetUp ()
+
+      Assert.That(result.Length, Is.EqualTo(1));
+      Assert.That(result[0], Is.TypeOf(typeof(NotEmptyBinaryValidator)));
+      Assert.That(((NotEmptyBinaryValidator)result[0]).ValidationMessage, Is.Not.Null);
+
+      validationMessageStub.Setup(_ => _.ToString()).Returns("Stub Message");
+      Assert.That(((NotEmptyBinaryValidator)result[0]).ValidationMessage.ToString(), Is.EqualTo("Stub Message"));
+    }
+
+    [Test]
+    public void GetPropertyValidator_ForBinaryProperty_WithCustomMessage ()
+    {
+      var propertyInformation = PropertyInfoAdapter.Create(typeof(Customer).GetProperty(nameof(Person.Photograph)));
+      _attribute.ErrorMessage = "CustomMessage";
+
+      var result = _attribute.GetPropertyValidators(propertyInformation, _validationMessageFactoryStub.Object).ToArray();
+
+      Assert.That(result.Length, Is.EqualTo(1));
+      Assert.That(((NotEmptyBinaryValidator)result[0]).ValidationMessage, Is.InstanceOf<InvariantValidationMessage>());
+      Assert.That(((NotEmptyBinaryValidator)result[0]).ValidationMessage.ToString(), Is.EqualTo("CustomMessage"));
+    }
+
+    [Test]
+    public void GetPropertyValidator_ForCollectionProperty_WithDefaultMessage ()
+    {
+      var propertyInformation = PropertyInfoAdapter.Create(typeof(Customer).GetProperty(nameof(Customer.ShippingAddresses)));
+      var validationMessageStub = new Mock<ValidationMessage>();
+      _validationMessageFactoryStub
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(It.IsAny<NotEmptyCollectionValidator>(), propertyInformation))
+          .Returns(validationMessageStub.Object);
+
+      var result = _attribute.GetPropertyValidators(propertyInformation, _validationMessageFactoryStub.Object).ToArray();
+
+      Assert.That(result.Length, Is.EqualTo(1));
+      Assert.That(result[0], Is.TypeOf(typeof(NotEmptyCollectionValidator)));
+      Assert.That(((NotEmptyCollectionValidator)result[0]).ValidationMessage, Is.Not.Null);
+
+      validationMessageStub.Setup(_ => _.ToString()).Returns("Stub Message");
+      Assert.That(((NotEmptyCollectionValidator)result[0]).ValidationMessage.ToString(), Is.EqualTo("Stub Message"));
+    }
+
+    [Test]
+    public void GetPropertyValidator_ForCollectionProperty_WithCustomMessage ()
+    {
+      var propertyInformation = PropertyInfoAdapter.Create(typeof(Customer).GetProperty(nameof(Customer.ShippingAddresses)));
+      _attribute.ErrorMessage = "CustomMessage";
+
+      var result = _attribute.GetPropertyValidators(propertyInformation, _validationMessageFactoryStub.Object).ToArray();
+
+      Assert.That(result.Length, Is.EqualTo(1));
+      Assert.That(((NotEmptyCollectionValidator)result[0]).ValidationMessage, Is.InstanceOf<InvariantValidationMessage>());
+      Assert.That(((NotEmptyCollectionValidator)result[0]).ValidationMessage.ToString(), Is.EqualTo("CustomMessage"));
     }
```

Comment:
I blieve these tests only covers one of the 3 branches for the collection validator. could you please verify and either explain what I'm missing or add the missing tests?